src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -252,6 +284,11 @@ public Task SendAsync(Message message)
                     "Cannot schedule messages in the past");
             }
 
+            if (this.isViaSender && Transaction.Current != null)
+            {
+                throw new ServiceBusException(false, $"{nameof(ScheduleMessageAsync)} method is not supported in a Via-Sender.");
```

Comment:
Should exception message be more specific as per _why_ it's not supported? There are two constraints, send-via and transaction.